virttest/libvirt_xml/other/idmap_xml.py

```diff
@@ -0,0 +1,101 @@
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Copyright Red Hat
+#
+#   SPDX-License-Identifier: GPL-2.0
+#
+#   Author: smitterl@redhat.com
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+from virttest import virsh
+from virttest.libvirt_xml import accessors, base, xcepts
+
+
+class VMIDMapXML(base.LibvirtXMLBase):
+    """
+    idmap xml class of vmxml
+
+    Example:
+      <idmap>
+        <uid start='0' target='1000' count='10'/>
+        <gid start='0' target='1000' count='10'/>
+      </idmap>
+    """
+
+    __slots__ = ("uid", "gid", "uids", "gids")
+
+    def __init__(self, virsh_instance=virsh):
+        accessors.XMLElementDict(
+            property_name="uid",
+            libvirtxml=self,
+            forbidden=None,
+            parent_xpath="/",
+            tag_name="uid",
+        )
+        accessors.XMLElementDict(
+            property_name="gid",
+            libvirtxml=self,
+            forbidden=None,
+            parent_xpath="/",
+            tag_name="gid",
+        )
+        accessors.XMLElementList(
```

Comment:
Seems `uid` `gid` could be replaced by newly added `uids` and `gids`. If you're not going to use `uid` `gid`, I'll remove these and update the test in tp-libvirt(Seems I wrote the only case uses these 2 attrs). The current change is ok to me.

Reply:
@chloerh Those two attributes already existed. I added the list attributes. I don't want to remove the existing attributes assuming that if they already existed, I don't want to touch them for backward compatibility.

Reply:
Actually I just want to make sure that the 2 old attributes are not used by the newly added tests. It should not block this pr, I will open another one to remove them and update existing tests. (Since I found the old 2 attrs were added by me and seems not used by other tests). Their functions overlap, that's the reason I'd like to remove the old ones. But it doesn't affect the current pr.

Reply:
@chloerh cool, can you approve the pr?

Reply:
Approved.